libstuff/ResourceMonitorThread.cpp

```diff
@@ -0,0 +1,26 @@
+#include "ResourceMonitorThread.h"
+#include "libstuff/libstuff.h"
+#include "format"
+#include <cmath>
+
+thread_local uint64_t ResourceMonitorThread::startTime;
+thread_local double ResourceMonitorThread::cpuStartTime;
+
+void ResourceMonitorThread::before(){
+    startTime = STimeNow();
+    cpuStartTime = SGetCPUUserTime();
+}
+
+void ResourceMonitorThread::after(){
+    const uint64_t threadEndTime = STimeNow() - startTime;
```

Comment:
```suggestion
    const uint64_t endTime = STimeNow() - startTime;
```

Either that or rename `startTime` to `threadStartTime`